numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -28,6 +28,29 @@ extern "C" {
  * PyInt -> PyLong
  */
 
+
+/*
+ * Copy of Python private function _PyLong_AsInt
+ * This is useful in replacing PyInt_Check.
```

Comment:
What had tripped me originally, was that the comment seems a bit misleading since `PyInt_Check` really only ensures `long` on the C-side.

I find `npy3_compat` a bit weird. Do we actually need this anywhere within NumPy itself? Maybe we should just put it into f2py?

Reply:
@eric-wieser replaced a lot of the non-controversial uses of `PyInt_Check` with `PyLong_Check`, but there are still about 10 uses outside of f2py that are more complicated and need looking at. I don't *know* that the new function is needed without going through them, conversion to C int is probably rare, but I think it is good to complete the set of `PyLong_As*` functions provided in the Python public API.

Reply:
OK, lets just put it in here! Should we modify the comment a bit to indicate that it is not strictly speaking an npy3 compat issue?